src/pudl/transform/eia861.py

```diff
@@ -194,79 +622,779 @@ def service_territory(raw_dfs, tfr_dfs):
+
+    """
+    raw_mergers = tfr_dfs["mergers_eia861"].copy()
+
+    # No data tidying required
+
+    ###########################################################################
+    # Transform Values:
+    # * Turn ownership column from single-letter code to full ownership category.
+    # * Retain preceeding zeros in zip codes
+    ###########################################################################
+
+    transformed_mergers = (
+        raw_mergers.assign(
+            entity_type=lambda x: x.entity_type.map(pc.ENTITY_TYPE_DICT),
+            merge_zip_5=lambda x: pudl.helpers.zero_pad_zips(x.merge_zip_5, 5),
+            merge_zip_4=lambda x: pudl.helpers.zero_pad_zips(x.merge_zip_4, 4)
+        )
+    )
+
+    # No duplicates to speak of but take measures to check just in case
+    _check_for_dupes(transformed_mergers, 'Mergers', [
+                     "utility_id_eia", "state", "report_date"])
+
+    # Organize col headers for output
+    transformed_mergers = (
+        pudl.helpers.organize_cols(
+            transformed_mergers, ['utility_id_eia', 'utility_name_eia',
+                                  'state', 'report_date']
+        )
+    )
+
+    tfr_dfs["mergers_eia861"] = transformed_mergers
+    return tfr_dfs
+
+
+def net_metering(tfr_dfs):
+    """
+    Transform the EIA 861 Net Metering table.
+
+    Args:
+        tfr_dfs (dict): A dictionary of transformed EIA 861 DataFrames, keyed by table
+            name. It will be mutated by this function.
+
+    Returns:
+        dict: A dictionary of transformed EIA 861 dataframes, keyed by table name.
+
+    """
+    idx_cols = [
+        'utility_id_eia',
+        'state',
+        'report_date',
+        'balancing_authority_code_eia'
+    ]
+
+    # Pre-tidy clean specific to sales table
+    raw_nm = tfr_dfs["net_metering_eia861"].copy()
+
+    ###########################################################################
+    # Tidy Data:
+    ###########################################################################
+
+    logger.info("Tidying the EIA 861 Net Metering table.")
+    # Normalize by customer class
+    tidy_nm = _tidy_class_dfs(raw_nm, 'Net Metering',
+                              idx_cols, CUSTOMER_CLASSES, 'customer_class')
+    # maybe delete these cols from excel so there aren't duplicates?
+    tidy_nm = tidy_nm.drop(columns='customers')
+    # Normalize by fuel class
+    tidy_nm = _tidy_class_dfs(tidy_nm, 'Net Metering',
+                              idx_cols + ['customer_class'], FUEL_CLASSES, 'fuel_class')
+
+    # No duplicates to speak of but take measures to check just in case
+    _check_for_dupes(tidy_nm, 'Net Metering',
+                     idx_cols + ['customer_class', 'fuel_class'])
+
+    # No transformation needed
+
+    # Organize col headers for output
+    tidy_nm = pudl.helpers.organize_cols(
+        tidy_nm, idx_cols + ['utility_name_eia', 'customer_class', 'fuel_class'])
```

Comment:
this is also nbd, but I assume we are going to end up stripping out the `utility_name_eia` when we normalize these tables. Is this added here it just for readability?

Reply:
Yes